### ISSUE TITLE
Error 400: Google Advanced Protection Blocks Mozilla Thunderbird OAuth Login (policy_enforced)

### ISSUE THREAD

**Issue by hirale** [thunderbird/thunderbird-android#8256]
### Checklist

- [X] I have used the search function to see if someone else has already submitted the same bug report.
- [X] I will describe the problem with as much detail as possible.

### App version

8.0b1

### Where did you get the app from?

Google Play

### Android version

12

### Steps to reproduce

1.  Add Account
2.  Type google workspace email address with custom domain name, admin@custom-domain.com
3.  Sign in with google

### Expected behavior

Account added as same as other Gmail accounts

### Actual behavior

![image](https://github.com/user-attachments/assets/d37f0dfa-4e74-4812-9628-7d8beb1913d5)

Error 400: policy_enforced
Request details: login_hint=admin@custom-domain.com response_type=code redirect_uri=net.thunderbird.android.beta:/oauth2redirect state=xxxxxxxxxxx code_challenge_method=S256 hd=custom-domain.comnonce=xxxxxxxxxxx client_id=xxxxxxxxxxx  code_challenge=xxxxxxxxxxx  access_type=offline scope=https://mail.google.com/ flowName=GeneralOAuthFlow

**Comment by kewisch** [comment]
Thanks for reporting. There is not a lot we can do about this right now, as there is no application process for Google's advanced protection afaik. Once we have the website updated for the release I'll see if I can find a way, this makes it a bit easier to explain.

**Comment by wynnchel** [comment]
> Thanks for reporting. There is not a lot we can do about this right now, as there is no application process for Google's advanced protection afaik. Once we have the website updated for the release I'll see if I can find a way, this makes it a bit easier to explain.

How was the application process handled for the desktop client? As the desktop client is accepted by the Google Advanced Protection Program, someone at Mozilla was assumably in contact with Google.

Best
Wynnchel

**Comment by fcsager** [comment]
I'm sorry I can't provide a more permanent solution, but I think I have a viable workaround.

I removed myself from Google's Advanced Protection program earlier this evening to try Thunderbird-Android. After completing the removal process, I signed in to the Gmail account I wished to use in Thunderbird-Android. Once I found my account was working within Thunderbird-Android, I re-enrolled in Google's Advanced Protection program.

So far, I am still able to send and receive messages within Thunderbird-Android. I don't know how long this result will persist, but I will report when it fails to continue working.

**Comment by kewisch** [comment]
Thank you, I added this as a workaround to the FAQ which will be published soon.

**Comment by lethedata** [comment]
Rather than completely disabling Advanced Protection you can set the App ID (client_id) to Trusted. (I'm guessing this is the "exception" mentioned in the FAQ)

Playing with the scopes by replicating the desktop client[^1] didn't help with anything and unfortunately the error message doesn't really give any additional details on what scopes it's trying to access.

[^1]: App ID: 406964657835-pq5dckrt5j6ijkicfp2jf4vdhesncql6.apps.googleusercontent.com

**Comment by trevormosey** [comment]
With the changes in 49109ad5cbe08025426d139f91938078f30a2b29 this is still not fixed unfortunately, the same policy enforcement error occurs when attempting to enroll APP accounts in Thunderbird for Android.

**Comment by kewisch** [comment]
Indeed, us changing the keys won't magically allows Google Advanced Protection. Now would be the time to find a contact at Google in the right department to allowlist the new keys we use.
